backend/app/models.py

```diff
@@ -44,5 +58,285 @@ class Document(models.Model):
+        new word.
+
+        Note: the method is not case sensitive and words always return lowercase.
+
+        :param target_word: Single word to search for in the document's text
+        :return: a Python Counter() object with {associated_word: occurrences}
+        """
+        target_word = target_word.lower()
+        word_count = Counter()
+        check = False
+        text = self.get_tokenized_text()
+
+        for word in text:
+            if check:
+                word_count[word] += 1
+                check = False
+            if word == target_word:
+                check = True
+        return word_count
+
+    def get_word_windows(self, search_terms, window_size=2):
+        """
+        Finds all instances of `word` and returns a counter of the words around it.
+        window_size is the number of words before and after to return, so the total window is
+        2*window_size + 1.
+
+        This is not case sensitive.
+
+        :param search_terms: String or list of strings to search for
+        :param window_size: integer representing number of words to search for in either direction
+        :return: Python Counter object
+        """
+
+        if isinstance(search_terms, str):
+            search_terms = [search_terms]
+
+        search_terms = set(i.lower() for i in search_terms)
+
+        counter = Counter()
+
+        for text_window in windowed(self.get_tokenized_text(), 2 * window_size + 1):
+            if text_window[window_size] in search_terms:
+                for surrounding_word in text_window:
+                    if surrounding_word not in search_terms:
+                        counter[surrounding_word] += 1
+
+        return counter
+
+    def get_word_freq(self, word):
+        """
+        Returns the frequency of appearance of a word in the document
+
+        :param word: str to search for in document
+        :return: float representing the portion of words in the text that are the parameter word
+        """
+        word_frequency = self.get_count_of_word(word) / self.get_word_count()
+        return word_frequency
+
+    def get_word_freqs(self, words):
+        """
+        A helper method for retrieving the frequencies of a given set of words within a Document.
+
+        :param words: a list of strings.
+        :return: a dictionary of words keyed to float frequencies.
+        """
+        word_frequencies = {word: self.get_count_of_word(word) / self.get_word_count() for word in words}
+        return word_frequencies
+
+    def get_part_of_speech_tags(self):
+        """
+        Returns the part of speech tags as a list of tuples. The first part of each tuple is the
+        term, the second one the part of speech tag.
+
+        Note: the same word can have a different part of speech tags.
+
+        :return: List of tuples (term, speech_tag)
+        """
+
+        if not self.part_of_speech_tags:
+            pos_tags = nltk.pos_tag(nltk.word_tokenize(self.text))
```

Comment:
I think this `nltk.word_tokenize(self.text)` is performing the tokenization we should assume occurs in `get_tokenized_text`. Can we therefore instead reference something like `JSON.loads(self.tokenized_text)`?